### PR TITLE
sys-freebsd/*10.1*: Changed to use tarball of upstream and bump EAPI=5.

### PR DESCRIPTION
**Pull request by nigoro**
Changed to use tarball of upstream.
@mgorny, You will be able to clean dev space ;-)

Bump to EAPI=5.
QA notice fix.

### CODE REVIEW

File: eclass/freebsd.eclass [hunk 1]
```diff
@@ -153,9 +162,23 @@ freebsd_src_unpack() {
...
 	else
-		for f in ${A} ; do
-			[[ ${f} == *.tar.* ]] && unpack ${f}
-		done
+		if version_is_at_least 10.0 ${RV} ; then
+			local tarball="freebsd-src-${MY_PV}.tar.xz"
+			local topdir="usr/src/"
+			local extractlist
+			for i in ${EXTRACTONLY} ; do
+				extractlist+=" ${topdir}${i}"
```
Comment (mgorny) [inline eclass/freebsd.eclass]: I suggest using an array here.

File: eclass/freebsd.eclass [hunk 2]
```diff
@@ -153,9 +162,23 @@ freebsd_src_unpack() {
...
+			done
+			ebegin "Unpacking parts of ${tarball} to ${WORKDIR}"
+			pushd "${WORKDIR}" > /dev/null
```
Comment (mgorny) [inline eclass/freebsd.eclass]: ||die

File: eclass/freebsd.eclass [hunk 3]
```diff
@@ -153,9 +162,23 @@ freebsd_src_unpack() {
...
+			# https://bugs.gentoo.org/show_bug.cgi?id=550734
+			xzdec -c "${DISTDIR}/${tarball}" | tar -xpf - --strip-components=2 ${extractlist} 2> /dev/null || die "tar extract command failed"
+			popd > /dev/null
```
Comment (mgorny) [inline eclass/freebsd.eclass]: ||die

Also, you can alternatively use:

```
cd "${WORKDIR}" || die
...
cd - || die
```

File: eclass/freebsd.eclass [hunk 4]
```diff
@@ -153,9 +162,23 @@ freebsd_src_unpack() {
...
+			local tarball="freebsd-src-${MY_PV}.tar.xz"
+			local topdir="usr/src/"
+			local extractlist
+			for i in ${EXTRACTONLY} ; do
+				extractlist+=" ${topdir}${i}"
+			done
+			ebegin "Unpacking parts of ${tarball} to ${WORKDIR}"
+			pushd "${WORKDIR}" > /dev/null
+			# https://bugs.gentoo.org/show_bug.cgi?id=550734
+			xzdec -c "${DISTDIR}/${tarball}" | tar -xpf - --strip-components=2 ${extractlist} 2> /dev/null || die "tar extract command failed"
```
Comment (mgorny) [inline eclass/freebsd.eclass]: `xzdec` doesn't seem like a safe choice. How about `tar -xJpf`? Does that work with BSD tar?

File: profiles/thirdpartymirrors [hunk 5]
```diff
@@ -7,7 +7,7 @@ debian		http://ftp.au.debian.org/debian http://ftp.at.debian.org/debian http://f
...
...eBSD.org/pub/FreeBSD/ ftp://ftp.ar.FreeBSD.org/pub/FreeBSD/ ftp://ftp.au.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.au.FreeBSD.org/pub/FreeBSD/ ftp://ftp.at.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.at.FreeBSD.org/pub/FreeBSD/ ftp://ftp.br.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.br.FreeBSD.org/pub/FreeBSD/ ftp://ftp.ca.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.ca.FreeBSD.org/ ftp://ftp.cn.FreeBSD.org/pub/FreeBSD/ ftp://ftp.cz.FreeBSD.org/pub/FreeBSD/ ftp://ftp.dk.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.dk.FreeBSD.org/pub/FreeBSD/ ftp://ftp.ee.FreeBSD.org/pub/FreeBSD/ ftp://ftp.fr.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.fr.FreeBSD.org/pub/FreeBSD/ ftp://ftp.de.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.de.FreeBSD.org/pub/FreeBSD/ ftp://ftp.gr.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.gr.FreeBSD.org/pub/FreeBSD/ ftp://ftp.hk.FreeBSD.org/pub/FreeBSD/ ftp://ftp.is.FreeBSD.org/pub/FreeBSD/ ftp://ftp.id.FreeBSD.org/pub/FreeBSD/ ftp://ftp.ie.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.ie.FreeBSD.org/pub/FreeBSD/ ftp://ftp.it.FreeBSD.org/pub/FreeBSD/ ftp://ftp.jp.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.jp.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.kr.FreeBSD.org/pub/FreeBSD/ ftp://ftp1.us.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.us.FreeBSD.org/pub/FreeBSD/
```
Comment (mgorny) [inline profiles/thirdpartymirrors]: I'd appreciate if you didn't mix too many unrelated changes in one PR.

**Comment by nigoro** [comment]
I've updated freebsd.eclass.
I dropped xzdec. And, I changed to tar -xJpf.

Before you merge this request, please merge PR https://github.com/gentoo/gentoo-portage-rsync-mirror/pull/139 .

**Comment by mgorny** [comment]
Merged, thanks.

**Comment by mgorny** [comment]
And cleaned up the dev space :).
